src/fleet/azext_fleet/custom.py

```diff
@@ -0,0 +1,161 @@
+# --------------------------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for license information.
+# --------------------------------------------------------------------------------------------
+
+import os
+import re
+
+from knack.log import get_logger
+from knack.util import CLIError
+
+from azure.cli.core.commands.client_factory import get_subscription_id
+from azure.cli.core.util import sdk_no_wait
+
+from azext_fleet._client_factory import CUSTOM_MGMT_FLEET
+from azext_fleet._resourcegroup import get_rg_location
+from azext_fleet._helpers import print_or_merge_credentials
+
+
+logger = get_logger(__name__)
+
+
+def create_fleet(cmd,
+                 client,
+                 resource_group_name,
+                 name,
+                 dns_name_prefix=None,
+                 location=None,
+                 tags=None,
+                 no_wait=False):
+    FleetHubProfile = cmd.get_models(
+        "FleetHubProfile",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    Fleet = cmd.get_models(
+        "Fleet",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    if dns_name_prefix is None:
+        subscription_id = get_subscription_id(cmd.cli_ctx)
+        # Use subscription id to provide uniqueness and prevent DNS name clashes
+        name_part = re.sub('[^A-Za-z0-9-]', '', name)[0:10]
+        if not name_part[0].isalpha():
+            name_part = (str('a') + name_part)[0:10]
+        resource_group_part = re.sub('[^A-Za-z0-9-]', '', resource_group_name)[0:16]
+        dns_name_prefix = f'{name_part}-{resource_group_part}-{subscription_id[0:6]}'
+
+    fleetHubProfile = FleetHubProfile(dns_prefix=dns_name_prefix)
+    rg_location = get_rg_location(cmd.cli_ctx, resource_group_name)
+    if location is None:
+        location = rg_location
+
+    fleet = Fleet(
+        location=location,
+        tags=tags,
+        hub_profile=fleetHubProfile
+    )
+
+    return sdk_no_wait(no_wait, client.begin_create_or_update, resource_group_name, name, fleet)
+
+
+def update_fleet(cmd,
+                 client,
+                 resource_group_name,
+                 name,
+                 tags=None):
+    FleetPatch = cmd.get_models(
+        "FleetPatch",
+        resource_type=CUSTOM_MGMT_FLEET,
+        operation_group="fleets"
+    )
+    fleetPatch = FleetPatch(tags=tags)
+    return client.update(resource_group_name, name, None, fleetPatch)
+
+
+def show_fleet(cmd,  # pylint: disable=unused-argument
+               client,
+               resource_group_name,
+               name):
+    logger.info('Getting fleet: %s for resource group: %s', name, resource_group_name)
```

Comment:
Info level log is not output by default (need extra option `--verbose`). May use `print()` (to stdout, not recommended) or `logger.warning()` (to stderr).

Reply:
removing these logs, they were for dev purposes